todoman/ui.py

```diff
@@ -251,8 +259,18 @@ def detailed(self, todo):
         return rv
 
     def format_date(self, date):
+        """
+        :param datetime.date date: a datetime object
+        Returns date in the following format:
+        * if date == today or tomorrow: "Today" or "Tomorrow"
+        * else: return a string representing that date
+        * if no date is supplied, it returns empty_date
+        """
         if date:
-            rv = date.strftime(self.date_format)
+            if date in self.special_dates:
```

Comment:
Since `date` is a datetime (I know, misleading variable name), you need to use `date.date()` here, otherwise `special_dates` is never used.